From d154b3aa24edf897173953fe70f35f27efb909db Mon Sep 17 00:00:00 2001
From: Boerje Sewing <Boerje.Sewing@who-ing.de>
Date: Mon, 29 Jan 2018 14:43:05 +0100
Subject: [PATCH] dts: src: fix ksz8863 configuration.

* ksz8863 specific dts resides now in separate file
* reorganise setupfw dts to meet pfc style
* delete unnecessary/double dts stuff

Signed-off-by: Boerje Sewing <Boerje.Sewing@who-ing.de>
---
 arch/arm/boot/dts/wago-src-ksz8863.dtsi | 116 ++------------------------------
 arch/arm/boot/dts/wago-src-setupfw.dts  |  21 +-----
 arch/arm/boot/dts/wago-src.dtsi         |  71 ++-----------------
 3 files changed, 14 insertions(+), 194 deletions(-)

diff --git a/arch/arm/boot/dts/wago-src-ksz8863.dtsi b/arch/arm/boot/dts/wago-src-ksz8863.dtsi
index 82617f63..86c2290 100644
--- a/arch/arm/boot/dts/wago-src-ksz8863.dtsi
+++ b/arch/arm/boot/dts/wago-src-ksz8863.dtsi
@@ -7,23 +7,19 @@
  */
 
 / {
-	aliases {
-		mdio-gpio0 = &bitbang_mdio0;
-	};
-
-	ksz8863 {
+	ksz8863: ksz8863 {
 		compatible = "micrel,ksz8863";
 		pinctrl-names = "default";
 		pinctrl-0 = <&ksz8863_pins>;
 		status = "okay";
 
 		ksz,reset-switch;
-		ksz,reset-gpio = <&gpio3 10 GPIO_ACTIVE_HIGH>; /*L18 MII1_RX_CLKgpio3[10] / RMII2.EN-PHY */
-		ksz,disable-internal-ldo;
+		ksz,reset-gpio = <&gpio3 10 GPIO_ACTIVE_LOW>; /*L18 MII1_RX_CLKgpio3[10] / RMII2.EN-PHY */
+		//ksz,disable-internal-ldo;
 		ksz,mii-bus = <&bitbang_mdio0>;
 	};
 
-	swcfg_ksz8863 {
+	swcfg_ksz8863: swcfg_ksz8863 {
 		compatible = "swcfg,ksz8863";
 		status = "okay";
 
@@ -34,10 +30,7 @@
 		swcfg,vlans = <16>;
 	};
 
-	dsa@0 {
-                /* set status to 'okay' to enable dsa */
-                status = "okay";
-
+	dsa: dsa@0 {
 		compatible = "marvell,dsa";
 		#address-cells = <2>;
 		#size-cells = <0>;
@@ -57,12 +50,14 @@
 				reg = <1>;
 				label = "ethX1"; /* This is X2100 */
 				phy-handle = <&phy1>;
+				phy-pwrdown;
 			};
 
 			port@1 {
 				reg = <2>;
 				label = "ethX2"; /* This is X2102 */
 				phy-handle = <&phy2>;
+				phy-pwrdown;
 			};
 
 			port@2 {
@@ -72,100 +67,3 @@
 		};
 	};
 };
-
-&am33xx_pinmux {
-
-	cpsw_default_pins: pinmux_cpsw_default_pins {
-		pinctrl-single,pins = <
-			0x040 (PIN_OUTPUT_PULLDOWN | MUX_MODE3) /* gpmc_a0.rmii2_txen */
-			0x050 (PIN_OUTPUT_PULLDOWN | MUX_MODE3) /* gpmc_a4.rmii2_txd1 */
-			0x054 (PIN_OUTPUT_PULLDOWN | MUX_MODE3) /* gpmc_a5.rmii2_txd0 */
-			0x068 (PIN_INPUT_PULLDOWN  | MUX_MODE3) /* gpmc_a10.rmii2_rxd1 */
-			0x06C (PIN_INPUT_PULLDOWN  | MUX_MODE3) /* gpmc_a11.rmii2_rxd0 */
-			0x074 (PIN_INPUT_PULLDOWN  | MUX_MODE3) /* gpmc_wpn.rmii2_rxer */
-			0x108 (PIN_INPUT_PULLDOWN  | MUX_MODE1) /* mii1_col.rmii2_refclk */
-			0x088 (PIN_INPUT_PULLDOWN  | MUX_MODE2) /* gpmc_csn3.rmii2_crs_dv */
-		>;
-	};
-
-	cpsw_sleep_pins: pinmux_cpsw_sleep_pins {
-		pinctrl-single,pins = <
-			0x040 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x050 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x054 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x068 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x06C (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x074 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x108 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-			0x088 (PIN_INPUT_PULLDOWN | MUX_MODE7)
-		>;
-	};
-
-	ksz8863_pins: pinmux_ksz8863_pins {
-		pinctrl-single,pins = <
-			0x130 (PIN_OUTPUT_PULLUP | MUX_MODE7) /* L18 MII1_RX_CLK      7 GPIO3_10->EN-PHY->Enable_Switch */
-		>;
-	};
-
-	gpio_bitbang_mdio_pins: pinmux_gpio_bitbang_mdio_pins {
-		pinctrl-single,pins = <
-			/* MDIO */
-			0x148 (PIN_INPUT_PULLUP | SLEWCTRL_FAST | MUX_MODE7)	/* M17: mdio.gpio0_0 (mdio_data), SLEWCTRL_FAST?? */
-			0x14c (PIN_OUTPUT_PULLUP | MUX_MODE7)			/* M18:  mdc.gpio0_1 (mdio_clk)                   */
-		>;
-	};
-
-	gpio_bitbang_mdio_sleep_pins: pinmux_gpio_bitbang_mdio_sleep_pins {
-		pinctrl-single,pins = <
-			/* MDIO reset value */
-			0x148 (PIN_INPUT_PULLDOWN 	| MUX_MODE7)	/* mdio.gpio0_0 */				//m17
-			0x14c (PIN_INPUT_PULLDOWN 	| MUX_MODE7)	/* mdc.gpio0_1   */				//m18
-		>;
-	};
-
-};
-
-&mac {
-	pinctrl-names = "default";
-	pinctrl-0 = <&cpsw_default_pins>;
-	pinctrl-1 = <&cpsw_sleep_pins>;
-	status = "okay";
-
-	mac_control = <0x18021>; /* force 100MBit full-duplex */
-	active_slave = <1>;
-};
-
-&bitbang_mdio0 {
-       status = "okay";
-       pinctrl-names = "default", "sleep";
-       pinctrl-0 = <&gpio_bitbang_mdio_pins>;
-       pinctrl-1 = <&gpio_bitbang_mdio_sleep_pins>;
-       gpios = <&gpio0 1 0	/* 0: mdc  */
-	        &gpio0 0 0>;   	/* 1: mdio */
-
-       phy1: ethernet-phy@1 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <1>;
-       };
-
-       phy2: ethernet-phy@2 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <2>;
-       };
-};
-
-/*
- * Use a fixed phy for emac0.
- * that's rgmii1
- */
-&cpsw_emac1 {
-	phy-mode = "rmii";
-	fixed-link {
-		speed = <100>;
-		full-duplex;
-	};
-};
-
-&phy_sel {
-	rmii-clock-ext;
-};
diff --git a/arch/arm/boot/dts/wago-src-setupfw.dts b/arch/arm/boot/dts/wago-src-setupfw.dts
index 2d01f45..75cce2fd 100644
--- a/arch/arm/boot/dts/wago-src-setupfw.dts
+++ b/arch/arm/boot/dts/wago-src-setupfw.dts
@@ -8,7 +8,7 @@
 /dts-v1/;
 
 #include "wago-src.dtsi"
-#include "wago-src-ksz8863.dtsi"
+#include "wago-devconf-desc.dtsi"
 
 /*
  *
@@ -23,25 +23,6 @@
  *
  */
 
-/*
- * The devconf variable is a 16bit value.
- * Data is stored in little endian format.
- *
- * 0x1fe |---------------------------------------|
- *       |          DEVCONF Byte 0 (Low Byte)    |
- *       | ------------------------------------- |
- *       |          DEVCONF Byte 1 (High Byte)   |
- * 0x200 |---------------------------------------|
- *
- *   15  14  13  12  11  10  9  8  7  6  5  4  3  2  1  0
- *    |   |   |   |   |   |  |  |  |  |  |  |  |  |  |  |
- *    x   x   x   x   x   x  x  x  x  x  x  x  |  |  |  PROFIBUS_SL
- *                                             |  |  CAN
- *                                             |  RS232_485
- *                                             ETHERNET_IP
- *   x: reserved for future use
- *
- */
 
 / {
 	target_info {
diff --git a/arch/arm/boot/dts/wago-src.dtsi b/arch/arm/boot/dts/wago-src.dtsi
index 5af8d10..db50f44 100644
--- a/arch/arm/boot/dts/wago-src.dtsi
+++ b/arch/arm/boot/dts/wago-src.dtsi
@@ -5,8 +5,9 @@
  * it under the terms of the GNU General Public License version 2 as
  * published by the Free Software Foundation.
  */
-#include "am33xx.dtsi"
 #include <dt-bindings/interrupt-controller/irq.h>
+#include "am33xx.dtsi"
+#include "wago-src-ksz8863.dtsi"
 
 /* Varianten beim SRC
  *
@@ -60,18 +61,6 @@
 		regulator-max-microvolt = <3300000>;
 	};
 
-	ksz8863 {
-		compatible = "micrel,ksz8863";
-		pinctrl-names = "default";
-		pinctrl-0 = <&ksz8863_pins>;
-		status = "okay";
-
-		ksz,reset-switch;
-		ksz,reset-gpio = <&gpio3 10 GPIO_ACTIVE_HIGH>; /*L18 MII1_RX_CLKgpio3[10] / RMII2.EN-PHY */
-		ksz,disable-internal-ldo;
-		ksz,mii-bus = <&bitbang_mdio0>;
-	};
-
 	wsysinit_init {
 		compatible = "wago,sysinit";
 		add-sysfs-entries;
@@ -116,56 +105,6 @@
 
 	};
 
-	swcfg_ksz8863 {
-		compatible = "swcfg,ksz8863";
-		status = "okay";
-
-		swcfg,mii-bus = <&bitbang_mdio0>;
-		swcfg,alias = "switch0";
-		swcfg,cpu_port = <2>;
-		swcfg,ports = <3>;
-		swcfg,vlans = <16>;
-	};
-
-	dsa@0 {
-		/* set status to 'okay' to enable dsa */
-		status = "okay";
-		compatible = "marvell,dsa";
-		#address-cells = <2>;
-		#size-cells = <0>;
-
-		/* interrupts = <10>; --> not supported from the driver! */
-
-		/* master netdevice (eth0) */
-		dsa,ethernet = <&mac>; /* TODO: should we use 'cpsw_emac1' instead? */
-		dsa,mii-bus = <&bitbang_mdio0>;
-
-		switch@0 {
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <16 0>;	/* MDIO address 16, switch 0 in tree */
-
-			port@0 {
-				reg = <1>;
-				label = "ethX1";
-				phy-handle = <&phy1>;
-				phy-pwrdown;
-			};
-
-			port@1 {
-				reg = <2>;
-				label = "ethX2";
-				phy-handle = <&phy2>;
-				phy-pwrdown;
-			};
-
-			port@2 {
-				reg = <3>;
-				label = "cpu";
-			};
-		};
-	};
-
 	/* LEDs DS and APP. Red color leds are driven by pwm to be able to
 	 * compensate temperature dependent yellow color.
 	 */
@@ -211,6 +150,10 @@
 
 };
 
+&ksz8863 {
+	ksz,disable-internal-ldo;
+};
+
 &elm {
 	status = "okay";
 };
@@ -277,8 +220,6 @@
 			0x178 ( MUX_MODE7 | PIN_OUTPUT		) /* D18 UART1_CTSN		7   GPIO0_12->DO_CPU_HLK1 */
 			0x17c ( MUX_MODE7 | PIN_OUTPUT		) /* D17 UART1_RTSN		7   GPIO0_13->DO_CPU_HLK2 */
 			0x1b4 ( MUX_MODE2 | PIN_INPUT		) /* D14 XDMA_EVENT_INTR1       2   tclkin->CNT-FAN->Counter_input */
-			/* FIXME: for testing purposes configure count in as gpio */
-			// 0x1b4 ( MUX_MODE7 | PIN_INPUT		) /* D14 XDMA_EVENT_INTR1	7   GPIO0_20->CNT-FAN->Counter_input */
 		>;
 	};
 
-- 
2.7.4

